includes/data-stores/class-wc-product-variation-data-store-cpt.php

```diff
@@ -165,6 +164,23 @@ public function update( &$product ) {
 	*/
 
 	/**
+	 * Generates a title with attribute information for a variation.
+	 * Products with one attribute will get a title of the form "Name - Value"
+	 * Products with 2+ attributes will get a title of the form "Name - Attribute: Value, Attribute: Value"
+	 *
+	 * @since 2.7.0
+	 * @param WC_Product
+	 * @return string
+	 */
+	protected function generate_product_title( $product ) {
+		$use_longform_title = apply_filters( 'woocommerce_product_variation_title_use_longform_style', count( $product->get_attributes() ) > 1, $product );
+		$title_base = apply_filters( 'woocommerce_product_variation_title_base', get_post_field( 'post_title', $product->get_parent_id() ), $product );
+		$separator = apply_filters( 'woocommerce_product_variation_title_separator', '&ndash;', $product );
```

Comment:
I don't think we need a separator filter. If we filter the return value, thats sufficient IMO